pkg/cluster/overridemtu.go

```diff
@@ -0,0 +1,94 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"github.com/coreos/ignition/v2/config/v3_2/types"
+	"github.com/openshift/installer/pkg/asset/ignition"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/machines/machineconfig"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/cluster/graph"
+)
+
+const (
+	IgnFilePath = "/etc/NetworkManager/dispatcher.d/30-eth0-mtu-3900"
+	IgnFileUser = "root"
+	IgnFileMode = 0555
+	IgnFileData = `#!/bin/bash
+
+if [ "$1" == "eth0" ] && [ "$2" == "up" ]; then
+    ip link set $1 mtu 3900
+fi`
+)
+
+func (m *manager) overrideEthernetMTU(g graph.Graph) error {
+	ignFile := ignition.FileFromString(IgnFilePath, IgnFileUser, IgnFileMode, IgnFileData)
+
+	// Extend Ignition config for master nodes
+
+	master := g.Get(&machine.Master{}).(*machine.Master)
+	master.Config.Storage.Files = append(master.Config.Storage.Files, ignFile)
+	data, err := ignition.Marshal(master.Config)
+	if err != nil {
+		return errors.Wrap(err, "failed to Marshal Ignition config")
+	}
+	master.File.Data = data
+
+	// Extend Ignition config for worker nodes by adding a
+	// MachineConfig manifest file to the bootstrap node's
+	// Ignition config.
+
+	ignConfig := types.Config{
+		Ignition: types.Ignition{
+			Version: types.MaxVersion.String(),
+		},
+		Storage: types.Storage{
+			Files: []types.File{ignFile},
+		},
+	}
+
+	rawExt, err := ignition.ConvertToRawExtension(ignConfig)
+	if err != nil {
+		return err
+	}
+
+	configs := []*mcv1.MachineConfig{
+		{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: mcv1.SchemeGroupVersion.String(),
+				Kind:       "MachineConfig",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "99-worker-mtu",
+				Labels: map[string]string{
+					"machineconfiguration.openshift.io/role": "worker",
```

Comment:
This is proving to be brittle as customers creating infra nodes are stripping "worker" labels.  And it's inconsistent with the way masters are being configured.  Could this be deployed to all machines in the same way?  Maybe we can use the `machine.openshift.io/region` label as we don't support deployment cross-region and every machine has this set to the same value?

Reply:
> Could this be deployed to all machines in the same way?

Maybe?  See your other comment below.  But as I understand, the label is dictated by the MachineConfigPool specification.
```
$ oc describe machineconfigpool worker
...
Spec:
  ...
  Machine Config Selector:
    Match Labels:
      machineconfiguration.openshift.io/role:  worker
  ...
```
So without the "worker" label, the `99-worker-mtu` MachineConfig would not be included in the `rendered-worker` MachineConfig, and thus wouldn't be applied to worker nodes.

Reply:
Fair enough, so it's a higher level issue and a reliance on having only master and worker roles.  Sucks when customers break that expectation but, shouldn't delete labels from machines and nodes :D